soc: arm: nxp_imx: rt*: don't enable MBEDTLS by default

bbilas opened the pull request:
See each commit message.

Signed-off-by: Bartosz Bilas <bartosz.bilas@hotmail.com>

bbilas committed:
soc: arm: nxp_imx: rt*: don't enable MBEDTLS by default

There is no reason to enable any cryptographic library
(no matter if it's mbedtls or tinycrypt) by default.
This allows using e.g NETWORKING (which selects ENTROPY_GENERATOR)
on RT* SOCs without MBEDTLS module enabled.

Signed-off-by: Bartosz Bilas <bartosz.bilas@hotmail.com>

bbilas commented:
@danieldegrasse @dleach02 gentle ping :)

bbilas commented on soc/arm/nxp_imx/rt/Kconfig.defconfig.series:
```diff
@@ -159,14 +159,6 @@ choice SEGGER_SYSVIEW_SECTION
-#
-config MBEDTLS
-#config TINYCRYPT
-	default y if CSPRING_ENABLED
```
Can't we just add the  `TINYCRYPT` or `MBEDTLS` dependencies for `CSPRING_ENABLED` option (didn't test it)? 
This way we can leave this option enabled by default.

<pre>
config CSPRING_ENABLED
#       bool "Cryptographically secure RNG functions enabled"
        bool
        default y
        depends on ENTROPY_HAS_DRIVER
        depends on ENTROPY_GENERATOR
	depends on MBEDTLS || TINYCRYPT
</pre>

dleach02 replied:
Yes, this looks like it might be better solution to how to manage CSPRING_ENABLED

bbilas replied:
Unfortunately we cannot do that because of dependency loop.

CSPRING_ENABLED (defined at subsys/random/Kconfig:80), with definition...

config CSPRING_ENABLED
	bool
	default y
	depends on ENTROPY_HAS_DRIVER && ENTROPY_GENERATOR && (MBEDTLS || TINYCRYPT)

...depends on MBEDTLS (defined at modules/mbedtls/Kconfig:17), with definition...

menuconfig MBEDTLS
	bool "mbed TLS Support" if !MBEDTLS_PROMPTLESS
	depends on 0
	help
	  This option enables the mbedTLS cryptography library.

(select-related dependencies: (CRYPTO_MBEDTLS_SHIM && CRYPTO) || (WIFI_ESP32 && DT_HAS_ESPRESSIF_ESP32_WIFI_ENABLED && !SMP && WIFI) || (JWT_SIGN_RSA && CSPRING_ENABLED && ) || (UPDATEHUB_DTLS && UPDATEHUB) || (OSDP_SC_ENABLED && CSPRING_ENABLED && OSDP) || (OPENTHREAD_MBEDTLS && NET_L2_OPENTHREAD && NETWORKING) || (NET_TCP_ISN_RFC6528 && NET_TCP && !NET_RAW_MODE && NETWORKING) || (WEBSOCKET_CLIENT && NETWORKING) || (NET_SOCKETS_SOCKOPT_TLS && NET_NATIVE && NET_SOCKETS && NETWORKING) || (FLASH_AREA_CHECK_INTEGRITY_MBEDTLS && ))

...depends on the choice symbol JWT_SIGN_RSA (defined at subsys/jwt/Kconfig:17), with definition...

config JWT_SIGN_RSA
	bool "Use RSA signature (RS-2[56](https://github.com/zephyrproject-rtos/zephyr/actions/runs/4609113742/jobs/8145794711?pr=55377#step:7:57))"
	select MBEDTLS
	depends on CSPRING_ENABLED && 

...depends again on CSPRING_ENABLED (defined at subsys/random/Kconfig:80))
FAILED: CMakeFiles/html
I've replaced that with enabling CSPRING when MBEDTLS or TINYCRYPT is available.

bbilas replied:
There is still dependency loop, I'm gonna take a look at it once again next week.

bbilas replied:
@danieldegrasse @mmahadevan108 @dleach02 It seems I've fixed that but it required more work than I thought... There were a lot of dependency loops but I hope everything was tested under CI. Please take a look once again.

bbilas commented:
@dleach02 Is there any conclusion from NXP side about that? It's been a while but is still not resolved.

bbilas committed:
random: make CSPRING dependent on entropy generator and crypto lib

As it's known that CSPRING needs an implementation of `ctr_drbg`
that either comes from MBEDTLS or TINYCRYPT in Zephyr
so let's make dependent it on entropy generator and mbetls or tinycrypt
lib.

Signed-off-by: Bartosz Bilas <bartosz.bilas@hotmail.com>

bbilas committed:
jwt: change select to depends on in Kconfig

It fixes the dependency loop of `CSRPING_ENABLED` option that
will be introduced in the upcoming commit that will add
a dependency MBEDTLS/TINYCRPY as well. While at it fix
the google_iot_mqtt sample by enabling missing TINYCRYPT module.

Signed-off-by: Bartosz Bilas <bartosz.bilas@hotmail.com>

bbilas committed:
mgmt: osdp: change select to depends on for MBEDTLS

It fixes the dependency loop of `CSRPING_ENABLED` option that
will be introduced in the upcoming commit that will add
a dependency MBEDTLS as well.

Signed-off-by: Bartosz Bilas <bartosz.bilas@hotmail.com>

mmahadevan108 commented:
@mrfuchs @sidcha do these changes look fine to you?

mmahadevan108 commented:
@dleach02, could you please re-review.

dleach02 commented on subsys/random/Kconfig:
```diff
@@ -82,6 +82,8 @@ config CSPRING_ENABLED
 	default y
 	depends on ENTROPY_HAS_DRIVER
+	depends on ENTROPY_GENERATOR
+	depends on MBEDTLS || TINYCRYPT
```
The problem I'm trying to figure out now is where should MBEDTLS || TINYCRYPT be set? If you try to run the /tests/crypto/rand32 test sample it now doesn't test the csrand functions because CONFIG_CSPRING_ENABLED doesn't get set.

bbilas replied:
Currently, the most suitable place to set `MBEDTLS/TINYCRYPT` is `prj.conf` within `tests/crypto/rand32` ( as it is done for `samples/net/cloud/google_iot_mqtt`). Nothing better does not come to my mind.

dleach02 replied:
so the reason it was being set in soc device tree was that some target SOCs didn't have enough resources (memory) to support mbedtls, hence the tinycrypt. It doesn't make sense to put it in the prj file because now you are making assumptions about the target in the application space unless the application specifically needs mbedtls over tinycrypt. 

Although an alternative is to have a board specific prj file but then it does still seem to go back to just defining it in the board defconfig file.

bbilas replied:
@dleach02 Unfortunately, I'm closing this PR as I will no longer have the opportunity to work with the RT10XX MCU. Feel free to go ahead with it if you want.
